model_analyzer/config/generate/base_model_config_generator.py

```diff
@@ -100,12 +103,10 @@ def _get_next_model_config(self):
         raise NotImplementedError
 
     def _last_results_increased_throughput(self):
-        max_throughput = self._get_last_results_max_throughput()
-        max_throughput_increased = all(
-            max_throughput is not None and t is not None and max_throughput > t
-            for t in self._curr_max_batch_size_throughputs)
-
-        return max_throughput_increased
+        if len(self._curr_max_batch_size_throughputs) >= 2:
```

Comment:
This is a pretty big change. It may be correct depending on how you have reorganized the rest of the code. I'll have to look closer.  I was hoping we wouldn't have to change all of the code like this to make it work.

Reply:
Okay! One reason for the change is that since we (almost) immediately update `self._curr_max_batch_size_throughputs` after the new measurement (or throughput) is observed, so we should not query `self._get_last_results_max_throughput()` for checking whether the latest throughput has increased (Plus, we need at least two measurements to decide that). I also fixed a small error in my code.